### ISSUE TITLE
Lodash modularized libs cause false positive for `import/extensions`

### ISSUE THREAD

**Issue by sindresorhus** [benmosher/eslint-plugin-import#378]
#### Config

```js
'import/extensions': [2, 'never'],
```

#### Code

```js
import template from 'lodash.template';
```

#### Result

```
✖   14:22  Unexpected use of file extension "template" for "lodash.template"  import/extensions
```

-

Not sure how this should be handled, but I definitely don't want to specify all the Lodash modules as extensions. Ideally, Lodash would have used proper slugged names like `lodash-template`...

-

// @jdalton

**Comment by jdalton** [comment]
If you have a rule against extensions you could have a list of what those extensions are (deny list).

**Comment by benmosher** [comment]
I think ideally we should resolve the imported file and ensure that _its_ extension is not being used. Directories with dots should be fine. 

It won't help in cases where the import can't be resolved, obviously. I think that's a substantial improvement.

**Comment by sindresorhus** [comment]
@benmosher 👍

**Comment by sindresorhus** [comment]
@jdalton The problem is that my rule is to deny all extensions, so listing them wouldn't work, but @benmosher's proposal sounds like a good solution.

**Comment by benmosher** [comment]
Fun fact: the rule already does what I suggested, probably as of 1.8.0. Looks like this PR is pinned to 1.7.0.

Upgrading your import plugin version should fix it?

**Comment by sindresorhus** [comment]
@benmosher I used 1.9.0.

**Comment by benmosher** [comment]
Weird. Are you using a custom resolver? There was a critical issue with custom resolver loading in 1.9.0, just published the fix via 1.9.1 this morning.

I suspect you aren't, though. Will look deeper into this, there must be a bug with the resolution.

**Comment by sindresorhus** [comment]
No custom resolver, but turns out I didn't have the dependency in `node_modules`...

**Comment by jdalton** [comment]
🤘
